### PR TITLE
fix(decommission): make sure that the rf is smaller than the number of nodes for tablets keyspaces

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
### Testing

- [ ]

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by ShlomiBalalis** [comment]
At the moment, replication factor changes are not supported with tablets (probably will only be supported in 6.1)
Will keep it as draft until then. CC @bhalevy

**Comment by bhalevy** [comment]
> At the moment, replication factor changes are not supported with tablets (probably will only be supported in 6.1) Will keep it as draft until then. CC @bhalevy

Right.  This is https://github.com/scylladb/scylladb/issues/16129

File: sdcm/cluster.py [hunk 1]
```diff
@@ -3135,6 +3160,12 @@ def wait_node_fully_start(self, verbose=True, timeout=3600):
         self.log.info('Waiting for native_transport to be ready')
         self.wait_native_transport()
 
+    @property
+    def is_tablets_enabled(self):
+        with self.remote_scylla_yaml() as scylla_yml:
+            current_experimental_features = scylla_yml.experimental_features
+            return "tablets" in current_experimental_features
```
Comment (bhalevy) [inline sdcm/cluster.py:3167]: This may change with https://github.com/scylladb/scylladb/issues/18875

File: sdcm/cluster.py [hunk 2]
```diff
@@ -4924,10 +4944,31 @@ def get_node_ip_list(verification_node):
...
+        for region in node_list_per_region:
+            datacenters.update({dc_names_by_region[region]: len(node_list_per_region[region])})
+        dc_to_decrease = dc_names_by_region[region_to_decrease]
+        self.log.debug("Number of nodes by datacenter %s", datacenters)
+        for keyspace in test_keyspaces:
+            replication_strategy = ReplicationStrategy.get(execution_node, keyspace)
+            if isinstance(replication_strategy, NetworkTopologyReplicationStrategy):
+                current_rf = replication_strategy.replication_factors
+                if datacenters[dc_to_decrease] == int(current_rf[dc_to_decrease]):
+                    datacenters[dc_to_decrease] -= 1
```
Comment (bhalevy) [inline sdcm/cluster.py:4968]: can it be 0?
If so, better raise exception

**Comment by pehala** [comment]
@bhalevy @fruch Do you feel this is still needed? If so, I can update the PR

**Comment by fruch** [comment]
> @bhalevy @fruch Do you feel this is still needed? If so, I can update the PR

I think that @yarongilor handled the situation already.

**Comment by pehala** [comment]
> I think that @yarongilor handled the situation already.

OK, I will close this PR

**Pull request closed without merging**